### DISCUSSION TITLE
v1.0.0-rc.1

### DISCUSSION THREAD

**Discussion by brooksmtownsend** [wasmCloud/wasmCloud#1855] [Announcements]
Hey everyone! Creating a thread here to talk about the 1.0 release candidate 1. Please use [wash-cli v0.27.0-alpha.2](https://github.com/wasmCloud/wasmCloud/releases/tag/wash-cli-v0.27.0-alpha.2) to test this release as it includes all new client libraries, tools, and pins to 1.0 rc.1.

## Known Issues/Limitations
- [x] We're actively updating our 1.0 release candidate documentation. Make sure to browse with the 1.0 version selected if you're using documentation on https://wasmcloud.com/. (As of writing this, the quickstarts still need some updates!)
- [x] The wadm version bundled in this release does not have support for creating or deleting configuration. If you specify configuration in your manifest, please ensure you create it manually beforehand with wash config put. @brooksmtownsend is planning on taking this.
- [x] The wash scale and wash start commands for components do not have support for specifying a configuration, this will be fixed in the next wash alpha release. https://github.com/wasmCloud/wasmCloud/pull/1889
- [x] The components in the wasmCloud quickstart are pinned to WASI logging / cloud versions @0.1.0, while the host is pinned to @0.2.0-draft. @brooksmtownsend is updating these templates now. (fixed in https://github.com/wasmCloud/wasmCloud/pull/1856)
- [x] The wasmCloud dashboard was not updated by wash v0.27.0-alpha.2, but it is updated in #1857 and will be released in the next alpha version of wash.
- [x] The TypeScript example included in our quickstart is failing to fully write the outgoing body, you should still see a response but the component might hang indefinitely. https://github.com/bytecodealliance/ComponentizeJS/issues/99. Fixed in https://github.com/wasmCloud/wasmCloud/pull/1911
- [x] If you previously ran a version of `wadm` and updated to this version, your streams may be out of date. We should provide https://github.com/wasmCloud/wasmCloud/issues/1740 in the next alpha version of wash. https://github.com/wasmCloud/wadm/pull/262 and wadm v0.11.0-alpha.3 include patches for these issues.

## Possible feature adds
Though the release candidate is meant to denote a line where no new features or breaking changes are implemented, I'd like to call out a couple of possible beneficial changes to make before we draw the line in the stability sand
- [x] [Refactor and streamlining of the keyvalue interfaces](https://github.com/WebAssembly/wasi-keyvalue/pull/41) is a significantly nicer-to-use API for keyvalue which is one of our core capabilities in wasmCloud. This would both make the component API nicer for examples and applications, and reduce our immediate maintenance burden with an adapter. @rvolosatovs and @thomastaylor312 are both interested parties here. #1859. Implemented in #1863

**Please let us know if there are any issues or rough edges that you run into while testing! We're available here and on the [wasmCloud Slack](https://slack.wasmcloud.com/).**

**Comment by rvolosatovs** [comment]
Refs #1859

**Comment by ricochet** [comment]
It looks like I have the pre-1.0 wadm when I install with this wash rc. I was expecting this to be bundled in with the rc and my wash up (thanks to https://github.com/wasmCloud/wasmCloud/pull/1853).

```powershell
# clean slate
wash drain all
rm -r ~/.wash

cargo install --git https://github.com/wasmCloud/wasmCloud --tag wash-cli-v0.27.0-alpha.2 --jobs 4 wash-cli

# start and host and check we're all good
wash up
wash get inventory

# follow quick start
rustup target add wasm32-wasi
wash new component hello --template-name hello-world-rust
cd hello
wash config put default-http address=127.0.0.1:8080
wash build

# errors with wadm
wash app deploy wadm.yaml

Could not put manifest to deploy Validation Error:
Should be able to parse object at: spec/components/ at index: 0
Please check for missing or incorrect elements

# check version
wash --version
wash 0.27.0-alpha.2
```

**Reply by lachieh** [reply]
You can specify the alpha of wadm with ` wash up --wadm-version v0.11.0-alpha.2`. Be sure wadm isn't running first with `wash down`

**Reply by brooksmtownsend** [reply]
🤔 You _should_ have the v0.11.0-alpha.2 wadm with the alpha of wash, I wonder if this is an issue relating to the above need of a wash drain for wadm streams

**Reply by ricochet** [reply]
I don't actually know if I needed to drain, I was just trying to get to a super-duper clean slate and didn't want to forget adding any context here for how I did that.

**Reply by brooksmtownsend** [reply]
Oh! Yes the validation error is a bug. It's fixed in https://github.com/wasmCloud/wasmCloud/pull/1856

**Reply by brooksmtownsend** [reply]
@ricochet if you can retry from `wash new component`, see how that works out for you

**Reply by ricochet** [reply]
Works!

**Comment by thomastaylor312** [comment]
`list_keys` for wasi:keyvalue is not yet implemented

**Reply by thomastaylor312** [reply]
Looks like @rvolosatovs added this in #1876

**Reply by rvolosatovs** [reply]
Yes, also closed https://github.com/wasmCloud/wasmCloud/issues/1743

**Comment by brooksmtownsend** [comment]
As far as small developer experience improvements, here's the list I was able to compile from my journey through all 4 of our quickstarts:

**wasmcloud**:
- [x] Similar to how we allow overwriting of component references for components, we should do the same things for providers https://github.com/wasmCloud/wasmCloud/pull/1909
- [x] The HTTP server provider still seems to fail the periodic health check, ensure we aren't msgpack serializing the httpserver health check https://github.com/wasmCloud/wasmCloud/issues/1925
- [x] The error message for linking to a component on the wrong interface is unintuitive and appears more like a panic, consider at least logging a nicer error on the host side https://github.com/wasmCloud/wasmCloud/issues/1927

**wash**:
- [x] Remove the `?` from `wash app history` output (the header) https://github.com/wasmCloud/wasmCloud/pull/1889
- [x] Left-align the "Host Inventory" header from wash get inventory https://github.com/wasmCloud/wasmCloud/pull/1889 
- [x] Update the dashboard version @lachieh https://github.com/wasmCloud/wasmCloud/pull/1890
- [x] Rename configuration block in wasmcloud.toml from `[actor]` to `[component]`, keeping the name backwards compatible for deserialization purposes. https://github.com/wasmCloud/wasmCloud/pull/1889
- [x] ~`wash down` should remove the pidfile for the host even if disconnected~
- [x] `wash link del` should have a nicer way of specifying the interface instead of `--namespace <namespace> --package <package>`. Strawman would be `wash link del <component_id> <wit>`. So instead of `wash link del component --namespace wasi --package http`, it would be `wash link del component wasi:http`. https://github.com/wasmCloud/wasmCloud/issues/1926
- [x] Enable websockets by default https://github.com/wasmCloud/wasmCloud/pull/1890

**wadm**:
- [x] Consider a nicer error message for fetching links when the only host in a lattice stops (fails with no responders, as expected. Not a huge deal)
- [x] Build the component ID based on the manifest name and the component name, not the sanitized reference (way too long) https://github.com/wasmCloud/wadm/pull/259 
- [x] Fix bug where when deploying a new manifest old components (just wasm components) are not stopped https://github.com/wasmCloud/wadm/pull/261
- [x] The wadm link scaler should react to the deleted event https://github.com/wasmCloud/wadm/pull/261
- [x] Rename the ID of the spreadscaler from `actor` to `component`

Folks should feel free to fix these if they see them, but I'm going to come back later today / this week to just knock a bunch of them out

**Reply by lachieh** [reply]
Could you add "enable websocket by default with option to disable" to the `wash` section

**Reply by brooksmtownsend** [reply]
Now that you fixed websockets, we need to make sure to remember to change docs

- [x] https://github.com/wasmCloud/wasmcloud.com/issues/370

**Reply by lachieh** [reply]
@brooksmtownsend should merge wasmcloud/wasmcloud.com#371 after the next wash RC and then we can close that too 👍🏻

**Comment by brooksmtownsend** [comment]
Just a note that for anyone updating wadm, the correct command to resolve issues involves recreating the mirror stream. I may try and patch this into wadm.
```
nats stream del wadm_mirror --force
nats kv del wadm_state host_default
```

The latter `kv del` I'm fixing with a backwards compatible deserialize in `wadm`.

This is because of the change to the wadm event topics, which previously listened for all events on `wasmbus.evt.*` but now listens on `wasmbus.evt.*.>`

**Reply by brooksmtownsend** [reply]
https://github.com/wasmCloud/wadm/pull/262 resolves this and with wadm v0.11.0-alpha.3 both of these issues will be fixed.

**Comment by ricochet** [comment]
Should the actor crate be renamed? https://github.com/wasmCloud/wasmCloud/tree/main/crates/actor

**Reply by brooksmtownsend** [reply]
Yes, probably to something like `wasmcloud_component_sdk`. It's not a huge issue for now as we don't use the crate in our examples, since it alters the way Rust looks compared to other components, but we should rename it and use it more

**Comment by brooksmtownsend** [comment]
All of the above fixes should be released into the next version of wash (v0.27.0-alpha.3).
- wasmCloud 1.0.0-rc.2, which includes the updated `wasi-keyvalue` implementations
- wadm v0.11.0-alpha.3, which includes bug fixes as listed above
- wash-lib v0.20.0-alpha.2, which includes bug fixes as listed above

**Reply by thomastaylor312** [reply]
wasmCloud 1.0.0-rc.2 and wadm v0.11.0-alpha.3 have been cut

**Reply by thomastaylor312** [reply]
wash-lib-v0.20.0-alpha.2 also has been cut

**Comment by brooksmtownsend** [comment]
- [x] As identified in https://github.com/wasmCloud/wasmCloud/pull/1878 we need to ensure that our providers are e2e implemented with tracing, which @vados-cosmonic is tackling
